clickhouse/base/string_utils.h

```diff
@@ -23,4 +26,37 @@ inline T FromString(const StringView& s) {
    return result;
 }
 
+inline const char* UUIDFormat() {
+   if constexpr (sizeof(long) == 8) {
+      return "%.8lx-%.4lx-%.4lx-%.4lx-%.12lx";
+   } else {
+      return "%.8llx-%.4llx-%.4llx-%.4llx-%.12llx";
+   }
+}
+
+template <>
+inline UUID FromString<UUID>(const std::string& s) {
+   uint64_t f1 = 0, f2 = 0, f3 = 0, s1 = 0, s2 = 0;
+   std::scanf(s.c_str(),UUIDFormat(), f1, f2, f3, s1, s2);
```

Comment:
1. `sscanf` maybe?
2. you should check the return code of the `sscanf`, since it may indicate that not all fields were read.
3. both overloads should share the code, `std::string`-overload should call `StringView`-overload
4. Please add some minor unit-tests for these `FromString` to ensure that implementation is correct. (both positive and error cases)